### ISSUE TITLE
Default easyrtcsidRegExp does not support express-session signed cookies

### ISSUE THREAD

**Issue by zacharysang** [priologic/easyrtc#431]
The default regex value of easyrtcsidRegExp is: `/^[a-z0-9_.-]{1,64}$/i`. However the session ids generated by `express-session` violate this pattern.

Here is a sample session id: `s%3AhYdr-V-d6B-2PSsAdnkD8aTNLJjV3vLu.Qilpnf0QIBMm1OCMoKpuBJjsx9f4yNv92EORI1TRIAA`

This violates the pattern in 2 ways:
1. Contains `%` character. This is caused by sanitization of `s:` -> `%3A`, which is used by express session as a prefix to all session ids (see [here](https://github.com/expressjs/session/issues/604))
2. Generated session id is too long (80 chars) whereas the limit of this pattern is 64 chars

**Comment by hthetiot** [comment]
@zacharysang I understand your issue, the problem I have with changing the default easirtcid reg exp because it is used on object properties and I think many EasyRTC user would not want to have something else then alphanumeric in the ID, Yet.

The current solution is to pass the option `easyrtcsidRegExp` see details here:
- https://github.com/priologic/easyrtc/blob/master/lib/easyrtc_default_options.js

@zacharysang can you provide new RegExp and I will consider patch over time.

**Comment by zacharysang** [comment]
Thanks for your response @hthetiot. I would like to clarify and ensure we are both talking about the easyrtc**s**id (not the normal easyrtcid). The value for easyrtcsidRegExp is `/^[a-z0-9_.%-]{1,100}$/i` (although this may not precisely describe the session ids generated by express-session).

While I understand wanting to avoid breaking changes for other users. I am not as clear on why the default behavior when enabling sessions results in invalid session ids. Does this mean that overriding the `easyrtcsidRegExp` option is required when enabling sessions? If so, is there a place in the docs that this requirement is stated?

**Comment by hthetiot** [comment]
Thank you @zacharysang, I miss understood "easyrtcsid" vs "easyrtcsid".

I'm willing to change the easyrtcsidRegExp I will do PR with `/^[a-z0-9_.%-]{1,100}$/i` for you to test.

**Comment by hthetiot** [comment]
see #435 @zacharysang

**Comment by hthetiot** [comment]
@zacharysang if you can test the branch `ugs/easyrtcsidRegExp-options` then i will merge if all good.

**Comment by zacharysang** [comment]
An update on this: 

The specified regex seems to be working well. I have had it running on a local project of mine for a couple weeks without issue (I started using this around when I reported the issue).

However, I would like to look a little more into the session id generator being used to see if we can get a regex that more closely matches (we may be able to restrict the sid to less than 100 chars, and there may be some other characters in the supported charset that are still missing from the pattern (although this would be surprising since they have not appeared so far)). 

I will have more time to look into this later in April after several school projects are submitted.
